src/Components/Web.JS/src/Platform/WebAssemblyResourceLoader.ts

```diff
@@ -12,14 +12,52 @@ export class WebAssemblyResourceLoader {
       credentials: 'include',
       cache: networkFetchCacheMode
     });
+
+    // While we can expect an ASP.NET Core hosted application to include the environment, other
+    // hosts may not. Assume 'Production' in that case.
+    const environment = bootConfigResponse.headers.get('Blazor_Environment') || 'Production';
     const bootConfig: BootJsonData = await bootConfigResponse.json();
     const cache = await getCacheToUseIfEnabled(bootConfig);
 
-    return new WebAssemblyResourceLoader(bootConfig, cache);
+    return new WebAssemblyResourceLoader(bootConfig, environment, cache);
   }
 
-  constructor (public readonly bootConfig: BootJsonData, private cacheIfUsed: Cache | null)
-  {
+  constructor (public readonly bootConfig: BootJsonData, public readonly applicationEnvironment: string, private cacheIfUsed: Cache | null) {
+  }
+
+  readConfigFilesAsync(): Array<ConfigResult>{
+    let results = new Array<ConfigResult>();
+
+    const configFiles =  this.bootConfig.config;
+    if (!configFiles) {
+      return results;
+    }
+
+    const defaultAppSettingsJson = 'appsettings.json';
```

Comment:
Just to check, do we definitely want to be case-sensitive like this? What are the chances that someone calls it `appSettings.json` or `AppSettings.json`? Is it desirable to make it work in these cases too?